### PR TITLE
add README

### PR DESCRIPTION
**Pull request by oliviamiller**
readme and also added one code change to continue the thread if an error occurs so that you don't have to restart the server.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,77 @@
-# Berry-IMU
\ No newline at end of file
+# Berry-IMU
```
Comment (randhid) [inline README.md:1]: Let's add the meta json too in this pr.

File: README.md [hunk 2]
```diff
@@ -1 +1,77 @@
...
+the i2c addresses `1c` and `6a`. 
+
+### Registry
+This module can be installed via the viam registry. 
+
+- Go to your robot's page on app.viam.com.
+- Click on the *Create Component* button in the Components section.
+- Search for the *berryimu* module and select it. 
+
+This will automatically install the module to your robot.
```
Comment (oliviamiller) [inline README.md]: Should I also have instructions on how to run locally?

File: README.md [hunk 3]
```diff
@@ -1 +1,77 @@
...
+        "calibrate": true,
+        "i2c_bus": "1",
+      },
+      "namespace": "viam-labs"
+      "type": "movement_sensor"
+      "model": "viam-labs:movement_sernsor:berryimu"
+            }
+        ]
+    }
+```
```
Comment (randhid) [inline README.md]: Is the the most updated way to configure modules? I was expecting a config that adds the modules and the components from that module into the config field

File: README.md (hunk 1 above)
Comment (oliviamiller) [reply README.md:1]: That gets generated when you upload the module right? will add

**Review by randhid — APPROVED** [review]
Double-check the configuration but looks good to me!

File: README.md (hunk 2 above)
Comment (randhid) [reply README.md]: No it's okay - maybe point to the docs for running a module locally

File: README.md (hunk 3 above)
Comment (oliviamiller) [reply README.md]: this is correct for the components section, ill also add the modules section to the sample to reduce confusion

**Review by randhid — APPROVED** [review]

**Comment by randhid** [comment]
LGTM but wait for someone from docs @npentrel feel free to delegate

File: meta.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "url": "https://github.com/viam-labs/berry-imu",
+    "description": "Implementation of the berryIMU version3 with i2c connection",
+    "models": [
+      {
+        "api": "rdk:component:movement_sensor",
+        "model": "viam-labs:movement_sensor:berryimu"
+      }
+    ],
+    "entrypoint": "run.sh"
+  }
```
Comment (randhid) [inline meta.json]:
```suggestion
  }
  
```

File: README.md [hunk 5]
```diff
@@ -1 +1,89 @@
...
\ No newline at end of file
+# Berry-IMU
+This module provides an implementation of the [BerryIMUv3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) using an i2c connection on ARM64 systems.
+
+## Setup
+
+###  Wiring
+Ensure i2c communication is enabled on your single board computer. Wire the SCL and SDA pins on the board to the corresponding pins on the IMU. 
+The IMU can be powered with the 3.3v pin. You can verify the i2c connection by running `i2cdetect -y 1` on your board. If the IMU is wired correctly, you will see 
+the i2c addresses `1c` and `6a`.
```
Comment (andf-viam) [inline README.md]: If desired, you could also link to the docs instructions for adding I2C support to a RPi. Maybe something like:

```suggestion
- Ensure i2c communication is enabled on your single board computer. For example, if you are using a Raspberry Pi, see [Enable communications protocols](https://docs.viam.com/installation/prepare/rpi-setup/#enable-communication-protocols).
- Wire the SCL and SDA pins on the board to the corresponding pins on the IMU. The IMU can be powered with the 3.3v pin.
- You can verify the i2c connection by running `i2cdetect -y 1` on your board. If the IMU is wired correctly, you will see the i2c addresses `1c` and `6a`. 
```

If you don't want to focus so much on the RPi usage case, feel free to disregard!

File: README.md [hunk 6]
```diff
@@ -1 +1,89 @@
...
+This module can be installed via the viam registry. 
+
+- Go to your robot's page on app.viam.com.
+- Click on the *Create Component* button in the Components section.
+- Search for the *berryimu* module and select it. 
+
+This will automatically install the module to your robot.
+
+
+You can also [run the module locally](https://docs.viam.com/registry/configure/#local-modules).
```
Comment (andf-viam) [inline README.md]: Nit: I would rely on the docs instructions, which also covers local modules on the same page. What do you think about this version, similar to the [agilex LIMO README](https://github.com/viam-labs/agilex):

```suggestion
### Add the `berryimu` module

To use this module, follow the instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry), and search for the `berryimu` movement sensor.
Click **Add module**, choose a name, and click **Create** to add the `berryimu` module and the modular movement sensor it provides.

You can also [run the module locally](https://docs.viam.com/registry/configure/#local-modules).
```

File: README.md [hunk 7]
```diff
@@ -1 +1,89 @@
...
+
+
+# Calibration
+The BerryIMU's magnetometer needs to be calibrated prior to using this module. Running the calibration script in 'src/calibration.py'
+will provide configuration values to correct for soft and hard iron distortions. 
+
+be sure to have the viam python sdk installed on your machine before running the calibration script:
+```
+pip install viam-sdk
+```
```
Comment (andf-viam) [inline README.md]:
````suggestion
To run the calibration script, you must install the Viam Python SDK on your single board computer:

```sh
pip install viam-sdk
```
````

File: README.md [hunk 8]
```diff
@@ -1 +1,89 @@
...
+
+Run the calibration script on your single board computer by calling `python3 calibration.py` in the `src` folder.
```
Comment (andf-viam) [inline README.md]:
````suggestion
Then, navigate to the `src` directory in the Python SDK, and run the calibration script using the following command:

```sh
python3 calibration.py
```
````

Bumped SBC mention up to "install Py SDK" step.

File: README.md [hunk 9]
```diff
@@ -1 +1,89 @@
...
+
+The script will prompt you to rotate the IMU in all directions. Make sure you are rotating the IMU 360 degrees in the X, Y, and Z directions 
+to ensure an accurate calibration. Keep rotating the IMU until the script outputs the calibration values. Copy and paste these values into your
+IMU component config in the viam app.
```
Comment (andf-viam) [inline README.md]:
```suggestion
IMU component config in the [Viam app](https://app.viam.com/robots).
```

File: README.md [hunk 10]
```diff
@@ -1 +1,89 @@
...
+
+# Attributes and Sample Config 
+The attributes for the module are as follows:
+   - `i2c_bus` (required): the name of the i2c bus wired to the IMU.
+   - `calibrate`: set to true to calibrate the magnetometer using the min and max values.
+   - `hard_iron_x_max`, `hard_iron_y_max`, `hard_iron_x_min`, `hard_iron_y_min`, `soft_iron_x_max`, `soft_iron_y_max`, `soft_iron_x_min`, `soft_iron_y_min`, (required when calibrate is `true`): values obtained from
+    calibration script.
```
Comment (andf-viam) [inline README.md]:
```suggestion
   - `i2c_bus`: (**Required**) The name of the i2c bus wired to the IMU.
   - `calibrate`: Set to true to calibrate the magnetometer using the min and max values.
   - `hard_iron_x_max`, `hard_iron_y_max`, `hard_iron_x_min`, `hard_iron_y_min`, `soft_iron_x_max`, `soft_iron_y_max`, `soft_iron_x_min`, `soft_iron_y_min`: (**Required** when calibrate is `true`) The calibration values obtained from running the `calibration.py` script.
```

File: README.md [hunk 11]
```diff
@@ -1 +1,89 @@
...
+        "hard_iron_y_max": 1953,
+        "soft_iron_x_max": 1069,
+        "hard_iron_x_min": -1953,
+        "hard_iron_x_max": 118,
+        "soft_iron_x_min": -1069,
+        "soft_iron_y_max": 1069,
+        "soft_iron_y_min": -1056,
+        "hard_iron_y_min": -1029
+        "calibrate": true,
+        "i2c_bus": "1",
```
Comment (andf-viam) [inline README.md]:
```suggestion
      "name": "myberryimu",
      "namespace": "rdk",
      "type": "movement_sensor",
      "model": "viam-labs:movement_sensor:berryimu",
      "attributes": {
        "hard_iron_y_max": 1953,
        "soft_iron_x_max": 1069,
        "hard_iron_x_min": -1953,
        "hard_iron_x_max": 118,
        "soft_iron_x_min": -1069,
        "soft_iron_y_max": 1069,
        "soft_iron_y_min": -1056,
        "hard_iron_y_min": -1029,
        "calibrate": true,
        "i2c_bus": "1"
```

**Review by andf-viam — APPROVED** [review]
Hi! LGTM mod some small clarifications and fixes. Taking over for Naomi as she is PTO today!

File: README.md [hunk 12]
```diff
@@ -1 +1,89 @@
-# Berry-IMU
\ No newline at end of file
+# Berry-IMU
+This module provides an implementation of the [BerryIMUv3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) using an i2c connection on ARM64 systems.
```
Comment (andf-viam) [inline README.md]: Sorry, one last comment! Let's expand this with a direct link to the module's registry page as well:

```suggestion
This module provides an implementation of the [BerryIMUv3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) using an i2c connection on ARM64 systems, and is [available from the Viam registry](https://app.viam.com/module/viam-labs/berryimu).
```

File: README.md (hunk 8 above)
Comment (andf-viam) [reply README.md]: Ah, sorry, you're right: `src` for this module, not the Python SDK. Thanks for catching! 👍

Comment (oliviamiller) [reply README.md]: no worries!!

**Pull request merged**